net/strongswan/files/ipsec.init

```diff
@@ -1,20 +1,327 @@
+	fi
+	echo "  keyexchange=$keyexchange" >> $FileConn
+}
+
+ConfigTunnel() {
+	local p2_proposal
+	local pfs_group
+	local encryption_algorithm
+	local authentication_algorithm
+
+	ConfigConn "$1"
+
+	# Specific for the tunnel part
+	config_get p2_proposal              "$1"           p2_proposal
+	config_get pfs_group                "$p2_proposal" pfs_group
+	config_get encryption_algorithm     "$p2_proposal" encryption_algorithm
+	config_get authentication_algorithm "$p2_proposal" authentication_algorithm
+
+	p2_proposal="${encryption_algorithm}-${authentication_algorithm}-${pfs_group}"
+
+	echo "  esp=$p2_proposal" >> $FileConn
+	echo "  ike=$Phase1Proposal" >> $FileConn
+	echo "  type=tunnel" >> $FileConn
+}
+
+ConfigTransport() {
+
+	ConfigConn "$1"
+
+	# Specific for the transport part
+	echo "  type=transport" >> $FileConn
+}
+
+ConfigRemote() {
+	local enabled
+	local gateway
+	local pre_shared_key
+	local authentication_method
+	local local_identifier
+	local remote_identifier
+
+	ConfigName=$1
+
+	config_get_bool enabled "$1" enabled 0
+	[[ "$enabled" == "0" ]] && return
+
+	EnabledRemotePeers=$((EnabledRemotePeers+1))
+
+	config_get gateway               "$1" gateway
+	config_get pre_shared_key        "$1" pre_shared_key
+	config_get authentication_method "$1" authentication_method
+	config_get local_identifier      "$1" local_identifier
+	config_get remote_identifier     "$1" remote_identifier
+
+	AuthenticationMethod=$authentication_method
+	LocalIdentifier=$local_identifier
+	RemoteIdentifier=$remote_identifier
+
+	RemoteGateway=$gateway
+	if [ "$RemoteGateway" = "any" ]; then
+		RemoteGateway="%any"
+		LocalGateway=`ip route get 1.1.1.1 | awk -F"src" '/src/{gsub(/ /,"");print $2}'`
+	else
+		LocalGateway=`ip route get $RemoteGateway | awk -F"src" '/src/{gsub(/ /,"");print $2}'`
+	fi
+	if [ -n "$LocalIdentifier" ]; then
+		echo -n "$LocalIdentifier " >> $FileSecrets
+	else
+		echo -n "$LocalGateway " >> $FileSecrets
+	fi
+	if [ -n "$RemoteIdentifier" ]; then
+		echo -n "$RemoteIdentifier " >> $FileSecrets
+	else
+		echo -n "$RemoteGateway " >> $FileSecrets
+	fi
+	echo ": PSK \"$pre_shared_key\"" >> $FileSecrets
+
+	Phase1Proposal=""
+	config_list_foreach "$1" p1_proposal ConfigPhase1
+	Phase1Proposal=`echo $Phase1Proposal | cut -b 2-`
+
+	config_list_foreach "$1" tunnel ConfigTunnel
+
+	config_list_foreach "$1" transport ConfigTransport
+
+	echo >> $FileConn
+}
+
+ConfigIpsec() {
+	local debug
+	local rtinstall_enabled
+	local routing_tables_ignored
+	local routing_table
+	local routing_table_id
+        local interface
+        local device_list
+
+	if [ ! -L /etc/ipsec.secrets ]; then
+		rm /etc/ipsec.secrets 2>/dev/null
+		ln -s $FileSecrets /etc/ipsec.secrets
+	fi
+
+	if [ ! -L /etc/strongswan.conf ]; then
+		rm /etc/strongswan.conf 2>/dev/null
+		ln -s $FileCommon /etc/strongswan.conf
+	fi
+
+	if [ ! -L /etc/ipsec.conf ]; then
+		rm /etc/ipsec.conf 2>/dev/null
+		ln -s $FileConn /etc/ipsec.conf
+	fi
+
```

Comment:
This will obviously break strongswan for anyone using it. Can we do something similar to openvpn, and support both manual config and UCI?